extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -278,7 +278,18 @@ export class WorkspaceService implements IWorkspaceService {
 	private async handleProjectProviderExtension(extension: vscode.Extension<any>): Promise<void> {
 		try {
 			if (!extension.isActive) {
-				await extension.activate();
+				const projectTypes = extension.packageJSON.contributes && extension.packageJSON.contributes.projects as string[];
```

Comment:
```suggestion
				const projectTypes = extension.packageJSON.contributes?.projects as string[] | undefined;
				if (!projectTypes) {
					return;
				}
```

Reply:
But also - this function only appears to be called by something that already is looking through the projects contribution. So you could probably reuse the value from that and pass it here? You'll need to handle the case when it's undefined though (activating all extensions)

Reply:
yeah I thought about passing the projType from `ensureProjectProviderLoaded()`, but then I still would need to do this anyway for the undefined (activate all extensions) case, so it didn't seem like there was much benefit to doing that.